sabnzbd/downloader.py

```diff
@@ -695,43 +700,43 @@ def run(self):
                     logging.info("Shutting down")
                     break
 
+            # If less data than possible was received then it should be ok to sleep a bit
+            if self.sleep_time:
+                if last_max_chunk_size > max_chunk_size:
+                    logging.debug("New max_chunk_size %d -> %d", max_chunk_size, last_max_chunk_size)
+                    max_chunk_size = last_max_chunk_size
+                elif last_max_chunk_size < max_chunk_size / 3:
+                    time_before = time.time()
+                    time.sleep(self.sleep_time)
+                    now = time.time()
+                    # Debugging code for v4 test release
+                    if now - time_before > self.sleep_time + 0.01:
+                        logging.debug("Slept %.5f seconds, sleep_time = %s", now - time_before, self.sleep_time)
+                    time_slept += now - time_before
+                    sleep_count += 1
+                    if sleep_count_start + 10 < now:
+                        logging.debug(
+                            "Slept %d times for an average of %.5f seconds the last %.2f seconds. sleep_time = %s",
+                            sleep_count,
+                            time_slept / sleep_count,
+                            now - sleep_count_start,
+                            self.sleep_time,
+                        )
+                        sleep_count_start = now
+                        sleep_count = 0
+                        time_slept = 0
+
+                last_max_chunk_size = 0
+
             # Use select to find sockets ready for reading/writing
             readkeys = self.read_fds.keys()
             if readkeys:
                 read, _, _ = select.select(readkeys, (), (), 1.0)
-
-                # Add a sleep if there are too few results compared to the number of active connections
-                if self.sleep_time:
-                    if can_be_slowed and len(read) < 1 + len(readkeys) / 10:
-                        time.sleep(self.sleep_time)
-
-                    # Initialize by waiting for stable speed and then enable sleep
-                    if can_be_slowed is None or can_be_slowed_timer:
-                        # Wait for stable speed to start testing
-
-                        if not can_be_slowed_timer and now > next_stable_speed_check:
-                            if BPSMeter.get_stable_speed(timespan=10):
-                                can_be_slowed_timer = now + 8
-                                can_be_slowed = 1
-                            else:
-                                next_stable_speed_check = now + _BPSMETER_UPDATE_DELAY
-
-                        # Check 10 seconds after enabling slowdown
-                        if can_be_slowed_timer and now > can_be_slowed_timer:
-                            # Now let's check if it was stable in the last 10 seconds
-                            can_be_slowed = BPSMeter.get_stable_speed(timespan=10)
```

Comment:
Can you also remove the get_stable_speed? This way we have 1 commit that has all that we can revert if needed.